api/v1beta1/azurecluster_validation.go

```diff
@@ -437,6 +437,16 @@ func validateNodeOutboundLB(lb *LoadBalancerSpec, old *LoadBalancerSpec, apiserv
 			fmt.Sprintf("Node outbound idle timeout should be between %d and %d minutes", MinLBIdleTimeoutInMinutes, MaxLoadBalancerOutboundIPs)))
 	}
 
+	for index, frontendIP := range lb.FrontendIPs {
```

Comment:
Although we intend to allow outboundIP only for nodeoutboundLB, there is nothing stopping the user from setting it in other places where publicIP is used (eg. AzureBastion). Should we add validations there to prevent the user from setting it?

Reply:
I notices PublicIP is used only in AzureBastion other than loadbalancer spec, do you know any other places where it is used or I have missed?